docs/user_guide/HOWTO/Configure_Ssh.rst

```diff
@@ -1,127 +1,121 @@
+   Examples using Hiera are provided for illustrative purposes.
 
 In Hiera:
 
 .. code-block:: yaml
 
    ssh::server::conf::port: 2222
-   ssh::server::conf::ciphers:
-   - 'chacha20-poly1305@openssh.com'
-   - 'aes256-ctr'
-   - 'aes256-gcm@openssh.com
-
-In Puppet:
-
-.. code-block:: puppet
-
-   include 'ssh::server'
-
-   # Alternative:
-   # if `ssh::enable_server: true`, this will also work
-   include 'ssh'
+   ssh::server::conf::passwordauthentication: false
 
 
 Managing Additional Settings with ``sshd_config``
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-To manage SSH server settings that are not managed by the SIMP ``ssh`` module,
-use the ``sshd_config`` resource from `augeasproviders_ssh`_.  This is what the
-SIMP ``ssh`` module uses internally to manage the ``/etc/ssh/sshd_config``
-file, and you can use it to set any options ``ssh::server::conf`` does not
-manage.
-
-For instance, to set the sshd ``LogLevel`` option to ``VERBOSE``:
-
-.. code-block:: puppet
-
-   # VERBOSE will log SSH key fingerprints used for logins
-   sshd_config { 'LogLevel' : value => 'VERBOSE' }
-
-
-Mixing ``ssh::server::conf`` and ``sshd_config``
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-Some SSH server configurations may require a combination of
-``ssh::server::conf`` (for options that SIMP manages) and ``sshd_config``
-resources (for additional options). The following example configures the
-``/etc/ssh/sshd_config`` keys ``GSSAPIAuthentication``, ``GSSAPIKeyExchange``,
-and ``GSSAPICleanupCredentials`` with a value of "**yes**":
-
-In Hiera:
+Starting with version **6.7.0** of the ``simp-ssh`` module, you can manage
+additional settings not explicitly mapped to ``ssh::server::conf`` parameters,
+using the ``ssh::server::conf::custom_entries`` parameter.  For example, to specify
+configuration for ``AllowAgentForwarding`` and ``AuthorizedPrincipalsCommand``
+``sshd`` settings, you would include Hiera such as the following:
 
 .. code-block:: yaml
 
-   # GSSAPIKeyExchange + GSSAPICleanupCredentials are managed via sshd_config
-   ssh::server::conf::gssapiauthentication: true
+  ssh::server::conf::custom_entries:
+    AllowAgentForwarding: "yes"
+    AuthorizedPrincipalsCommand: "/usr/local/bin/my_command"
 
-In Puppet:
+There are a few limitations with ``ssh::server::conf::custom_entries`` that
+need to be noted:
 
-.. code-block:: puppet
-
-   include 'ssh::server'
-
-   sshd_config {
-    default:
-      ensure => 'present',
-      value  => 'yes',
-    ;
-    # GSSAPIAuthentication is managed via `ssh::server::conf::gssapiauthentication`
-    ['GSSAPIKeyExchange', 'GSSAPICleanupCredentials']:
-      # use defaults
-    ;
-   }
+* *No setting validation*:
+    This parameter is **not validated**. Be careful to only specify settings
+    that are allowed for your particular SSH daemon and avoid duplicate
+    declaration of resources already specified. Invalid options may cause the
```

Comment:
I would change `resources` to `settings` in order to avoid confusion with a Puppet `resource`:
```
    declaration of settings already specified. Invalid options may cause the
```

Reply:
The settings are done with Puppet resources, so I think it may be appropriate to leave as is.

Reply:
I'll clarify.